### PR TITLE
address some code quality issues

### PR DESCRIPTION
**Pull request by danieleades**
a bunch of small, non-breaking, code quality improvements

### CODE REVIEW

File: sphinx/ext/viewcode.py [hunk 1]
```diff
@@ -68,10 +68,7 @@ def is_supported_builder(builder: Builder) -> bool:
         return False
     elif builder.name == 'singlehtml':
         return False
-    elif builder.name.startswith('epub') and not builder.config.viewcode_enable_epub:
-        return False
-    else:
-        return True
+    return not (builder.name.startswith('epub') and not builder.config.viewcode_enable_epub)
```
Comment (tk0miya) [inline sphinx/ext/viewcode.py]: Indeed, this is equivalent to the current code. But this changes the meaning of the code, I feel. I think it's not good to expand consecutive else-if block to such code.

**Review by tk0miya — CHANGES_REQUESTED** [review]
LGTM with nits.

**Comment by tk0miya** [comment]
It seems mypy emits warnings. Could you check them, please?

File: sphinx/ext/viewcode.py (hunk 1 above)
Comment (danieleades) [reply sphinx/ext/viewcode.py]: done

**Comment by danieleades** [comment]
> It seems mypy emits warnings. Could you check them, please?

mypy issues have been addressed, i believe. would you mind retriggering the CI for me @tk0miya?

**Comment by danieleades** [comment]
@tk0miya i *think* this should be good to merge now (I don't think this one is too controversial!)

File: sphinx/cmd/quickstart.py [hunk 2]
```diff
@@ -175,10 +175,7 @@ def _has_custom_template(self, template_name: str) -> bool:
               It will be removed in the future without deprecation period.
         """
         template = path.join(self.templatedir, path.basename(template_name))
-        if self.templatedir and path.exists(template):
-            return True
-        else:
-            return False
+        return bool(self.templatedir) and path.exists(template)
```
Comment (tk0miya) [inline sphinx/cmd/quickstart.py:178]: Why do you call `bool()` here?

File: sphinx/ext/apidoc.py [hunk 3]
```diff
@@ -181,23 +181,16 @@ def is_skipped_package(dirname: str, opts: Any, excludes: List[str] = []) -> boo
...
     if not path.exists(filename):
         # skip if the file doesn't exist
         return True
-    elif path.basename(filename).startswith('_') and not opts.includeprivate:
-        # skip if the module has a "private" name
-        return True
-    else:
-        return False
+    # skip if the module has a "private" name
+    return path.basename(filename).startswith('_') and not opts.includeprivate
```
Comment (tk0miya) [inline sphinx/ext/apidoc.py:193]: I feel this refactoring divides this if-blocks to an early return condition (`path.exists()`) and the main condition.  Do you think that the main condition of "is_skipped_module()` is to check the filename is private or not? IMO, it's one of the conditions. So they should not be separated.

File: sphinx/ext/autodoc/__init__.py [hunk 4]
```diff
@@ -2427,10 +2421,7 @@ def is_runtime_instance_attribute(self, parent: Any) -> bool:
         # An instance variable defined in __init__().
         if self.get_attribute_comment(parent, self.objpath[-1]):  # type: ignore
             return True
-        elif self.is_runtime_instance_attribute_not_commented(parent):
-            return True
-        else:
-            return False
+        return self.is_runtime_instance_attribute_not_commented(parent)
```
Comment (tk0miya) [inline sphinx/ext/autodoc/__init__.py:2424]: This is also.

File: sphinx/ext/autodoc/__init__.py [hunk 5]
```diff
@@ -2571,10 +2559,7 @@ def can_document_member(cls, member: Any, membername: str, isattr: bool, parent:
             return False
         elif inspect.isattributedescriptor(member):
             return True
-        elif not inspect.isroutine(member) and not isinstance(member, type):
-            return True
-        else:
-            return False
+        return not inspect.isroutine(member) and not isinstance(member, type)
```
Comment (tk0miya) [inline sphinx/ext/autodoc/__init__.py:2562]: This is also.

File: sphinx/ext/autodoc/type_comment.py [hunk 6]
```diff
@@ -26,13 +26,10 @@ def not_suppressed(argtypes: List[ast.AST] = []) -> bool:
     """Check given *argtypes* is suppressed type_comment or not."""
     if len(argtypes) == 0:  # no argtypees
         return False
-    elif len(argtypes) == 1 and ast_unparse(argtypes[0]) == "...":  # suppressed
+    return not (len(argtypes) == 1 and ast_unparse(argtypes[0]) == "...")  # suppressed
         # Note: To support multiple versions of python, this uses ``ast_unparse()`` for
         # comparison with Ellipsis.  Since 3.8, ast.Constant has been used to represent
         # Ellipsis node instead of ast.Ellipsis.
```
Comment (tk0miya) [inline sphinx/ext/autodoc/type_comment.py:32]: Could you move this comment to the above of the return statement?

File: sphinx/ext/autodoc/type_comment.py [hunk 7]
```diff
@@ -26,13 +26,10 @@ def not_suppressed(argtypes: List[ast.AST] = []) -> bool:
     """Check given *argtypes* is suppressed type_comment or not."""
     if len(argtypes) == 0:  # no argtypees
         return False
-    elif len(argtypes) == 1 and ast_unparse(argtypes[0]) == "...":  # suppressed
+    return not (len(argtypes) == 1 and ast_unparse(argtypes[0]) == "...")  # suppressed
```
Comment (tk0miya) [inline sphinx/ext/autodoc/type_comment.py:29]: The comment on the tail is no longer needed.

File: sphinx/util/inspect.py [hunk 8]
```diff
@@ -341,24 +335,18 @@ def isattributedescriptor(obj: Any) -> bool:
                                     WrapperDescriptorType)):
             # attribute must not be a method descriptor
             return False
-        elif type(unwrapped).__name__ == "instancemethod":
-            # attribute must not be an instancemethod (C-API)
-            return False
-        else:
-            return True
+        # attribute must not be an instancemethod (C-API)
+        return not type(unwrapped).__name__ == "instancemethod"
```
Comment (tk0miya) [inline sphinx/util/inspect.py:339]: This is also.

File: sphinx/util/inspect.py [hunk 9]
```diff
@@ -393,28 +381,19 @@ def iswrappedcoroutine(obj: Any) -> bool:
             # staticmethod, classmethod and partial method are not a wrapped coroutine-function
             # Note: Since 3.10, staticmethod and classmethod becomes a kind of wrappers
             return False
-        elif hasattr(obj, '__wrapped__'):
-            return True
-        else:
-            return False
+        return hasattr(obj, '__wrapped__')
```
Comment (tk0miya) [inline sphinx/util/inspect.py:384]: This is also.

File: sphinx/util/inspect.py [hunk 10]
```diff
@@ -435,11 +414,8 @@ def isgenericalias(obj: Any) -> bool:
     elif (hasattr(types, 'GenericAlias') and  # only for py39+
           isinstance(obj, types.GenericAlias)):  # type: ignore
         return True
-    elif (hasattr(typing, '_SpecialGenericAlias') and  # for py39+
-            isinstance(obj, typing._SpecialGenericAlias)):  # type: ignore
-        return True
-    else:
-        return False
+    return (hasattr(typing, '_SpecialGenericAlias') and  # for py39+
+            isinstance(obj, typing._SpecialGenericAlias))  # type: ignore
```
Comment (tk0miya) [inline sphinx/util/inspect.py:418]: This is also.

**Review by tk0miya — CHANGES_REQUESTED** [review]

File: sphinx/cmd/quickstart.py (hunk 2 above)
Comment (danieleades) [reply sphinx/cmd/quickstart.py:178]: mypy returns a false positive here without explicitly coercing to a bool. there might be a better way to address this. will investigate

File: sphinx/ext/apidoc.py (hunk 3 above)
Comment (danieleades) [reply sphinx/ext/apidoc.py:193]: hmm. If you wanted to return from within the if-else block in these sorts of cases, i'd be more inclined to go with a construction like the following:

```python
if (
    # skip if the file doesn't exist
    not path.exists(filename) or
    # skip if the module has a "private" name
    path.basename(filename).startswith('_') and not opts.includeprivate
):
    return True
else:
    return False
```

**Comment by AA-Turner** [comment]
Closed in favour of #10516.

A

**Pull request closed without merging**